plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/BaseSqlServerTypeMapping.java

```diff
@@ -836,6 +839,92 @@ private void testSqlServerDatetimeOffset(ZoneId sessionZone)
+                .execute(getQueryRunner(), session, sqlServerCreateAndInsert("test_sqlserver_datetimeoffset_historical_date"));
+    }
+
+    @Test
+    public void testSqlServerDatetimeOffsetHistoricalDatesRangeQuery()
+    {
+        // Tests the custom predicate push down controller for DATETIMEOFFSET types with values before and after 1583
+        List<String> dateTimeOffsetValues = List.of(
+                "'1400-01-01 00:00:00.1234567+00:00'",
+                "'1500-01-01 00:00:00.1234567+00:00'",
+                "'1582-12-31 23:59:59.9999999+00:00'",
+                "'1583-01-01 00:00:00+00:00'",
+                "'1583-01-01 00:00:00.1234567+00:00'",
+                "'1600-01-01 00:00:00.1234567+00:00'",
+                "'1700-01-01 00:00:00.1234567+00:00'",
+                "'1800-01-01 00:00:00.1234567+00:00'",
+                "'1900-01-01 00:00:00.1234567+00:00'");
+
+        try (TestTable table = new TestTable(onRemoteDatabase(), "test_sqlserver_datetimeoffset_historical_date_range_query", "(col0 datetimeoffset(7))", dateTimeOffsetValues)) {
+            assertQuery("SELECT count(*) FROM " + table.getName(), "SELECT 9");
+            assertQuery("SELECT count(*) FROM " + table.getName() + " WHERE col0 <= TIMESTAMP '1582-12-31 23:59:59.9999999+00:00'", "SELECT 3");
+            assertQuery("SELECT count(*) FROM " + table.getName() + " WHERE col0 >= TIMESTAMP '1583-01-01 00:00:00+00:00'", "SELECT 6");
+            assertQuery("SELECT count(*) FROM " + table.getName() + " WHERE col0 IN (TIMESTAMP '1582-12-31 23:59:59.9999999+00:00', TIMESTAMP '1583-01-01 00:00:00+00:00')", "SELECT 2");
+            assertQuery("SELECT count(*) FROM " + table.getName() + " WHERE col0 IN (TIMESTAMP '1583-01-01 00:00:00+00:00', TIMESTAMP '1600-01-01 00:00:00.1234567+00:00')", "SELECT 2");
+            assertQuery("SELECT count(*) FROM " + table.getName() + " WHERE col0 NOT IN (TIMESTAMP '1582-12-31 23:59:59.9999999+00:00', TIMESTAMP '1600-01-01 00:00:00.1234567+00:00')", "SELECT 7");
+            assertQuery("SELECT count(*) FROM " + table.getName() + " WHERE col0 NOT IN (TIMESTAMP '1583-01-01 00:00:00+00:00', TIMESTAMP '1600-01-01 00:00:00.1234567+00:00')", "SELECT 7");
+            assertQuery("SELECT count(*) FROM " + table.getName() + " WHERE col0 BETWEEN TIMESTAMP '1582-12-31 23:59:59.9999999+00:00' AND TIMESTAMP '1800-01-01 00:00:00.1234567+00:00'", "SELECT 6");
+            assertQuery("SELECT count(*) FROM " + table.getName() + " WHERE col0 BETWEEN TIMESTAMP '1583-01-01 00:00:00+00:00' AND TIMESTAMP '1800-01-01 00:00:00.1234567+00:00'", "SELECT 5");
+
+            assertThat(query("SELECT count(*) FROM " + table.getName()))
+                    .isFullyPushedDown();
+
+            assertThat(query("SELECT * FROM " + table.getName() + " WHERE col0 <= TIMESTAMP '1582-12-31 23:59:59.9999999+00:00'"))
```

Comment:
Can we also have an additional test case for `WHERE col0 <= TIMESTAMP '1990-01-01 00:00:00+00:00`